bayes_opt/target_space.py

```diff
@@ -116,6 +116,22 @@ def constraint_values(self):
         if self._constraint is not None:
             return self._constraint_values
 
+    @property
+    def mask(self):
+        mask = np.ones_like(self.target, dtype=bool)
+
+        # mask points that don't satisfy the constraint
+        if self._constraint is not None:
+            mask &= self._constraint.allowed(self._constraint_values)
+
+        # mask points that are outside the bounds
+        if self._bounds is not None:
+            within_bounds = np.all((self._bounds[:, 0] <= self._params) & 
```

Comment:
@perezed00 @bwheelz36 @leandrobbraga @fmfn 
What do you think, does it make sense trigger warning here if we find values outside pbounds?

Reply:
There are three ways for parameters to exist beyond the pbounds:
1. Existing pbounds are shrunk using `target_space.set_bounds(small_pbounds)`
2. They are manually (i.e., intentionally) added using `target_space.register(x, y)`
3. Logs are loaded that don't respect the pbounds. 

I don't think that #1 should produce a warning since this intended behavior. To tackle #2 and #3 without affecting #1, we could issue a warning when a point outside the bounds is **registered**, rather than found?

Reply:
That sounds good, would you mind implementing the warning + adding a test for it?

Reply:
This warning has been added and all warnings have been slightly updated for improved formatting.